task/gimlet-inspector/src/main.rs

```diff
@@ -0,0 +1,140 @@
+                                Some(regs),
+                            ),
+                            Err(SeqError::ServerRestarted) => (
+                                SequencerRegistersResponseV0::SequencerTaskDead,
+                                None,
+                            ),
+                            Err(_) => {
+                                // The SeqError type represents a mashing
+                                // together of all possible errors for all
+                                // possible sequencer IPC operations. The only
+                                // one we _expect_ here is ReadRegsFailed.
+                                (SequencerRegistersResponseV0::SequencerReadRegsFailed, None)
+                            }
+                        };
+                        let mut len =
+                            hubpack::serialize(&mut tx_data_buf, &resp)
+                                .unwrap_lite();
+                        if let Some(t) = trailer {
+                            tx_data_buf[len..len + t.len()].copy_from_slice(&t);
+                            len += t.len();
+                        }
+                        meta.size = len as u32;
+                    }
+                }
+
+                // With the response packet prepared, we may need to attempt
+                // sending more than once.
+                loop {
+                    match net.send_packet(
+                        SOCKET,
+                        meta,
+                        &tx_data_buf[0..(meta.size as usize)],
+                    ) {
+                        Ok(()) => break,
+                        // If `net` just restarted, immediately retry our send.
+                        Err(SendError::ServerRestarted) => continue,
+                        // If our tx queue is full, wait for space. This is the
+                        // same notification we get for incoming packets, so we
+                        // might spuriously wake up due to an incoming packet
+                        // (which we can't service anyway because we are still
+                        // waiting to respond to a previous request); once we
+                        // finally succeed in sending we'll peel any queued
+                        // packets off our recv queue at the top of our main
+                        // loop.
+                        Err(SendError::QueueFull) => {
+                            sys_recv_closed(
+                                &mut [],
+                                notifications::SOCKET_MASK,
+                                TaskId::KERNEL,
+                            )
+                            .unwrap_lite();
+                        }
+                        // These errors should be impossible if we're configured
+                        // correctly.
+                        Err(SendError::NotYours | SendError::InvalidVLan) => {
```

Comment:
out of curiosity: in these cases where we panic due to one of multiple unhandled errors, are we intentionally not indicating which it is in the panic to avoid putting more strings in the binary, or something? naively, it seems like it would be nice to know which of the two “things that shouldn’t happen” happened, on the off chance one of them does happen.

Reply:
> are we intentionally not indicating which it is in the panic to avoid putting more strings in the binary, or something?

Exactly. We get the line number, which I'm hoping will suffice in the event that this happens. (What we really _ought_ to do is divide up the error type to just the variants that can happen for a given operation.)

Reply:
yeah, that makes sense --- refining the error types so that each operation returns an enum of _just_ the variants that it can actually return seems like a good idea! i could, potentially, pick that up, given some pointers?